app/assets/javascripts/discourse/app/lib/composer/toolbar.js

```diff
@@ -174,6 +193,93 @@ export default class Toolbar extends ToolbarBase {
       active: ({ state }) => state.inItalic,
     });
 
+    const headingLabel = getButtonLabel("composer.heading_label", "H");
+    const unformattedHeadingIcon = headingLabel ? null : "discourse-text";
+    this.addButton({
+      id: "heading",
+      group: "fontStyles",
+      active: ({ state }) => {
+        if (!state || !state.inHeading) {
+          return false;
+        }
+
+        if (state.inHeading) {
```

Comment:
this should always be true right?

given we have this just before:

```js
if (!state || !state.inHeading) {
  return false;
}
```

So can the code just be: `return state.inHeadingLevel < 4` ?